### PR TITLE
Fees are now always reinvested (no more claim)

### PR DESCRIPTION
**Pull request by clemlak**
This PR changes the way fees are managed: they are now always reinvested in the pool and are not tracked anymore.
The PR also remove the tracking of the invariant growth.

By removing the claiming fees mechanism, the following Spearbit findings are now irrelevant:
- https://github.com/spearbit-audits/review-primitive/issues/17
- https://github.com/spearbit-audits/review-primitive/issues/13
- https://github.com/spearbit-audits/review-primitive/issues/8

### CODE REVIEW

File: contracts/Portfolio.sol [hunk 1]
```diff
@@ -541,19 +482,14 @@ abstract contract PortfolioVirtual is Objective {
 
         // -=- Apply Fee Saving Method -=- //
         {
-            // Fees are saved by incrementing the fee growth accumulator.
-            bool saved = _feeSavingEffects(args.poolId, iteration);
-
             // If the fees are not saved,
```
Comment (MrToph) [inline contracts/Portfolio.sol:485]: the comment should be changed, it only needs to talk about the reinvestment case.

File: contracts/Portfolio.sol [hunk 2]
```diff
@@ -541,19 +482,14 @@ abstract contract PortfolioVirtual is Objective {
...
-            if (!saved) {
-                if (_state.sell) {
-                    iteration.virtualX = nextIndependent;
-                } else {
-                    iteration.virtualY = nextIndependent;
-                }
+            if (_state.sell) {
+                iteration.virtualX = nextIndependent;
+            } else {
+                iteration.virtualY = nextIndependent;
```
Comment (MrToph) [inline contracts/Portfolio.sol:492]: fees are always reinvested now which will always increase the invariant, even above 0.

**Review by MrToph — COMMENTED** [review]
looks good to me besides the mentioned tracking error

File: contracts/libraries/FVMLib.sol [hunk 3]
```diff
@@ -37,7 +37,6 @@ bytes1 constant UNKNOWN = 0x00;
 bytes1 constant ALLOCATE = 0x01;
 bytes1 constant UNSET02 = 0x02;
 bytes1 constant DEALLOCATE = 0x03;
-bytes1 constant CLAIM = 0x04;
```
Comment (kmbarry1) [inline contracts/libraries/FVMLib.sol:40]: Perhaps it would be more consistent to leave a `bytes1 constant UNSET04 = 0x04;` here instead?

Comment (clemlak) [reply contracts/libraries/FVMLib.sol:40]: Yes, you're right.

**Pull request merged**